src/WinXP/apps/DisplayProperties/Pipes3DProperties.js

```diff
@@ -0,0 +1,224 @@
+import React, { useState, useContext } from 'react';
+import styled from 'styled-components';
+import Button from '../../../components/Button';
+import { Context as AppContext } from './../../../WinXP';
+
+import LegendFieldset from 'components/LegendFieldset';
+import RadioGroup from 'components/RadioGroup';
+import SelectInput from 'components/SelectInput';
+import Slider from 'components/Slider';
+
+import { DISPLAY_PROPERTIES } from '../../constants/actions';
+import {
+  MAX_SPEED,
+  MIN_SPEED,
+  getColorOptions,
+  getTexturePathOptions,
+  jointTypeOptions,
+} from './utils';
+
+function Pipes3DProperties({ onClose }) {
+  const appContext = useContext(AppContext);
+  const { displayProperties } = appContext.state;
+  const handleCancel = onClose;
+
+  /// Component state
+  const {
+    multiple,
+    texturePath,
+    joints,
+    interval,
+  } = displayProperties.screenSaversSettings.Pipes3D;
+  const [pipes3DState, setPipes3DState] = useState({
+    multiple,
+    texturePath,
+    joints,
+    interval,
+  });
+
+  const handleApply = () => {
+    appContext.dispatch({
+      type: DISPLAY_PROPERTIES,
+      payload: {
+        ...displayProperties,
+        screenSaversSettings: {
+          ...displayProperties.screenSaversSettings,
+          Pipes3D: pipes3DState,
+        },
+      },
+    });
+    onClose();
+  };
+
+  const handleChange = (value, field) => {
+    setPipes3DState(prev => ({ ...prev, [field]: value }));
+  };
+
+  const handleBooleanChange = (value, field) => {
+    let booleanValue = value === 'true' ? true : false;
+    setPipes3DState(prev => ({ ...prev, [field]: booleanValue }));
```

Comment:
No need to elaborate with the ternary, can be shortened to:
```suggestion
    setPipes3DState(prev => ({ ...prev, [field]: value === 'true' }));
```